devito/types/dimension.py

```diff
@@ -36,13 +36,11 @@ class Dimension(ArgProvider):
                                            |
                               ---------------------------
                               |                         |
-                       BasicDimension            DefaultDimension
+                       DerivedDimension            DefaultDimension
```

Comment:
maybe add all DImension subclasses to this plot?

Reply:
as the upper part of the docstring says, just the API classes